patches/server/0959-Bell-API.patch

```diff
@@ -0,0 +1,62 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lama06 <andreasprues36@gmail.com>
+Date: Sun, 22 Jan 2023 11:33:25 +0100
+Subject: [PATCH] Bell API
+
+== AT ==
+public net.minecraft.world.level.block.entity.BellBlockEntity resonating
+public net.minecraft.world.level.block.entity.BellBlockEntity resonationTicks
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBell.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBell.java
+index 6c5ed0fc9f58b9ec22194074a3fd69b387e3b270..d8fd1445482a8dd5804d297983023e497d05c498 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBell.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBell.java
+@@ -9,4 +9,48 @@ public class CraftBell extends CraftBlockEntityState<BellBlockEntity> implements
+     public CraftBell(World world, BellBlockEntity tileEntity) {
+         super(world, tileEntity);
+     }
++
++    // Paper start - Bell API
++    @Override
++    public void ring(org.bukkit.block.BlockFace direction) {
++        if (!direction.isCartesian()) throw new IllegalArgumentException("A bell can only ring in a cartesian direction");
++        if (direction.getModY() != 0) throw new IllegalArgumentException("A bell cannot ring up or down");
++        requirePlaced();
++        getTileEntity().onHit(net.minecraft.core.Direction.fromNormal(direction.getModX(), direction.getModY(), direction.getModZ()));
++    }
++
++    @Override
++    public boolean isRinging() {
++        return getTileEntity().shaking;
++    }
++
++    @Override
++    public int getHasBeenRingingFor() {
++        return getTileEntity().ticks;
++    }
++
++    @Override
++    public org.bukkit.block.BlockFace getRingingDirection() {
++        if (!isRinging()) return null;
++
++        return switch (getTileEntity().clickDirection) {
```

Comment:
You can use the CraftBlock.notchToBlockFace method to avoid that. Also it's a little bit strange that you have mapped the up/down direction since you explicitly blocked that in the ring method